lib/CodeGen/PagerandoBinning.cpp

```diff
@@ -47,16 +41,19 @@ class PagerandoBinning : public ModulePass {
   bool runOnModule(Module &M) override;
 
   void getAnalysisUsage(AnalysisUsage &AU) const override {
+    ModulePass::getAnalysisUsage(AU);
     AU.addRequired<MachineModuleInfo>();
     AU.addPreserved<MachineModuleInfo>();
     AU.setPreservesAll();
-    ModulePass::getAnalysisUsage(AU);
   }
 
 private:
-  // Map from free space -> Bin
-  typedef std::multimap<unsigned, Bin> BinMap;
-  BinMap Bins;
+  static constexpr unsigned DefaultBin = 0;
+  static constexpr unsigned BinSize = 4096;     // one page
+  static constexpr unsigned MinFreeSpace = 64;  // cache line (32 or 64 on ARM)
```

Comment:
`MinFreeSpace` should be smaller than 64 (maybe 4-8 bytes instead). There are plenty of functions smaller than 64 bytes, and we could fit many of them in those 64-byte gaps at the end.